databallpy/load_data/event_data/instat.py

```diff
@@ -0,0 +1,286 @@
+        .drop("team_id", axis=1)
+        .reset_index(drop=True)
+    )
+    away_players = df_players[df_players["team_id"] == metadata.away_team_id]
+    away_players = (
+        away_players.sort_values("starter", ascending=False)
+        .drop("team_id", axis=1)
+        .reset_index(drop=True)
+    )
+
+    metadata.home_formation = home_formation
+    metadata.away_formation = away_formation
+    metadata.home_players = home_players
+    metadata.away_players = away_players
+    # import pdb;pdb.set_trace()
+    return metadata
+
+
+def _load_event_data(event_data_loc: str, metadata: Metadata) -> pd.DataFrame:
+    """Function to load the event_data.json file, the events of the match.
+    Note: this function does ignore qualifiers for now
+
+
+    Args:
+        event_data_loc (str): location of the event_data.json file
+        metadata(Metadata): metadata of the match
+
+    Returns:
+        pd.DataFrame: event data of the match
+    """
+    EVENT_AND_OUTCOME_INSTAT_EVENTS = {
+        "Attacking pass accurate": ["pass", 1],
+        "Accurate key pass": ["pass", 1],
+        "Attacking pass inaccurate": ["pass", 0],
+        "Inaccurate key pass": ["pass", 0],
+        "Pass into offside": ["pass", 0],
+        "Successful dribbling": ["dribbling", 1],
+        "Unsuccessful dribbling": ["dribbling", 0],
+        "Dribbling": ["dribbling", np.nan],
+        "Crosses accurate": ["cross", 1],
+        "Accurate crossing from set piece with a shot": ["cross", 1],
+        "Accurate crossing from set piece": ["cross", 1],
+        "Accurate crossing from set piece with a goal": ["cross", 1],
+        "Crosses inaccurate": ["cross", 0],
+        "Cross interception": ["cross", 0],
+        "Inaccurate blocked cross": ["cross", 0],
+        "Inaccurate set-piece cross": ["cross", 0],
+        "Shot on target": ["shot", 1],
+        "Blocked shot": ["shot", 0],
+        "Wide shot": ["shot", 0],
+    }
+
+    with open(event_data_loc, "r") as f:
+        data = f.read()
+    event_data_json = json.loads(data)
+    events = event_data_json["data"]["row"]
+
+    result_dict = {
+        "event_id": [],
+        "type_id": [],
+        "event": [],
+        "period_id": [],
+        "minutes": [],
+        "seconds": [],
+        "player_id": [],
+        "team_id": [],
+        "outcome": [],
+        "start_x": [],
+        "start_y": [],
+        "datetime": [],
+        "instat_event": [],
+    }
+
+    start_time_period = {
+        1: metadata.periods_frames.loc[0, "start_datetime_ed"],
+        2: metadata.periods_frames.loc[0, "start_datetime_ed"]
+        + dt.timedelta(minutes=60),
+        3: metadata.periods_frames.loc[0, "start_datetime_ed"]
+        + dt.timedelta(minutes=110),
+        4: metadata.periods_frames.loc[0, "start_datetime_ed"]
+        + dt.timedelta(minutes=130),
+        5: metadata.periods_frames.loc[0, "start_datetime_ed"]
+        + dt.timedelta(minutes=150),
+    }
```

Comment:
This is used to create a datetime value for each event. The events only contain seconds relative to the start of the half. In syncing the first event is synced to the first frame of the tracking data for each half. An alternative would be to discard this start_time_period and only save the minutes. This will result in datetime objects on 1970/01/01 so I think this is more realistic, let me know if you disagree